src/background/modules/OmniboxSearch.js

```diff
@@ -171,7 +172,7 @@ export async function triggerOmnixboxSearch(text, disposition) {
                 copyToClipboard: true
             });
         } else if (emojiSearch.action === "emojipedia") {
-            const resultUrl = `https://emojipedia.org/search/?q=${encodeURIComponent(emojiText)}`;
+            const resultUrl = `https://emojipedia.org/search/?${new URLSearchParams({ q: emojiText })}`;
```

Comment:
Well should not we then better actually use `new URL()` for everything and append the param then?

Reply:
The `searchParams` property on the [URL interface](https://developer.mozilla.org/en-US/docs/Web/API/URL#instance_properties) is read only, so that would not be possible.

Reply:
Asked ChatGPT (via MDN AI help) and after some unhelpful parrot answers I got the reply of course: `searchParams` may be readonly, but `search` is not…

![image](https://github.com/rugk/awesome-emoji-picker/assets/11966684/5ec6785b-7e2a-429e-81c8-7feb52209354)

Reply:
Okay here the full response:
> Yes, you can encode URL parameters more nicely using the `URLSearchParams` and `URL` interfaces. In your code, you can replace the use of `encodeURIComponent` with the `URLSearchParams` constructor to achieve a cleaner and more readable solution.
>
>Here's an example of how you can combine the `URL` constructor with a base URL and `URLSearchParams` using the `append()` method to include a URI-encoded text:
>
>```js
const baseUrl = "https://emojipedia.org/search/";
const emojiText = "smiley face";

const params = new URLSearchParams();
params.append("q", emojiText);

const resultUrl = new URL(baseUrl);
resultUrl.search = params;

console.log(resultUrl.href);
// Output: "https://emojipedia.org/search/?q=smiley%20face"
```

>In this example, we first create a new `URLSearchParams` object and append the encoded `emojiText` using the `append()` method. Then, we create a new `URL` object with the base URL and assign the `params` to the `search` property of the `URL` object. Finally, we can access the encoded URL by accessing the `href` property of the `URL` object.
>
>By using the `URLSearchParams` and `URL` interfaces, you can easily construct URLs with encoded parameters without manually concatenating strings or using `encodeURIComponent`.
>
>I hope this helps! Let me know if you have any further questions.

- [URLSearchParams: toString() method](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/toString)
- [URLSearchParams: append() method](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/append)
- [URLSearchParams: set() method](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/set)
- [URLSearchParams: getAll() method](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/getAll)
- [URLSearchParams: URLSearchParams() constructor](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams)

Reply:
OK, this is a lot of additional code and I am not sure if it is any improvement over what we have now since you are not changing any other parts of the URL, but I am happy to make this change if you want...

For reference, as you may recall, I also have a couple of add-ons that use the Omnibox API and I used the `URLSearchParams` interface to construct the URL pramaters the same as I am proposing here. See my Bug Opener extension for example: https://github.com/tdulcet/Bug-Opener/blob/2ca42832227d806845e4df0391104488273ea4c8/background.js#L145-L150